nncf/quantization/algorithms/weight_compression/openvino_backend.py

```diff
@@ -179,62 +169,95 @@ class WeightNodeParams:
     compression_config = WeightCompressionConfig()
 
 
-def _int8_compress(
-    weight: np.ndarray, reduction_axes: Union[int, Tuple[int]]
+def _do_integer_quantization(
+    weight: np.ndarray, reduction_axes: Union[int, Tuple[int]], config: WeightCompressionConfig
 ) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
     """
-    Do unsigned int8 asymmetric weight compression - quantization to [0, 255] range.
+    The method quantizes the given weights to integer data type in accordance with the compression config.
+    The config defines a quantization mode:
+        INT8 mode refers to unsigned int8 asymmetric weight compression - quantization to [0, 255] range.
+        INT4_ASYM - to unsigned int4 asymmetric weight compression - quantization to [0, 15] range.
+        INT4_SYM - to unsigned int4 symmetric weight compression - quantization to [0, 15] range.
```

Comment:
I would mention here that there is a fixed zero point with value 8 for INT4_SYM.

Reply:
Reworked accordingly